doc/api.yml

```diff
@@ -1072,6 +1291,99 @@ components:
       properties:
         main_teacher:
           $ref: '#/components/schemas/Teacher'
+    Order:
+      type: string
+      enum:
+        - ASC
+        - DESC
+
+    ExamInfo:
+      type: object
+      properties:
+        title:
+          type: string
+        date:
+          type: string
+          format: date-time
+    Exam:
+      allOf:
+        - $ref: '#/components/schemas/ExamInfo'
+      properties:
+        id:
+          type: string
+        coefficient:
+          type: integer
+
+    ExamDetail:
+      type: object
+      properties:
+        id:
+          type: string
+        infos:
+          $ref: '#/components/schemas/ExamInfo'
+        participants:
+          type: array
+          items:
+            $ref: '#/components/schemas/Student'
+        student_grades:
+          type: array
+          items:
+            $ref: '#/components/schemas/StudentGrade'
```

Comment:
Why ? 

I mean, StudentGrade doesn't it combine student and grade.

But you have a list of participants that are students earlier, and here you will have duplicated infos about student.

I think you can combine with participants.

Reply:
Well in StudentGrade we have the student's id and ref and his grade on the specific exam. So why do we need the participants ? 
![Uploading image.png…]()